modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -342,25 +342,76 @@ PolygonConcentricCircleMeshGeneratorBase::PolygonConcentricCircleMeshGeneratorBa
     if (!_ring_block_ids.empty() &&
         _ring_block_ids.size() !=
             (_ring_intervals.size() + (unsigned int)(num_innermost_ring_layers != 1)))
-      paramError("ring_block_ids",
-                 "This parameter must have the appropriate size if it is provided. The size should "
-                 "be the same as the size of 'ring_intervals' if the innermost ring interval "
-                 "(including boundary layers) is unity; otherwise the size should be greater than "
-                 "the size of 'ring_intervals' by one. If 'quad_center_elements' is true, it is "
-                 "optional to only provide this parameter with the same size as 'ring_intervals'");
+    {
+      // Create an ostringstream for the debug information
+      std::ostringstream debug_info;
+      debug_info << "quad_center_elements is : " << (_quad_center_elements ? "true" : "false")
+                 << std::endl;
+      debug_info << "ring_block_ids size is : " << _ring_block_ids.size() << std::endl;
+      debug_info << "ring_intervals size is : " << _ring_intervals.size() << std::endl;
+      debug_info << "first element of ring_intervals is : " << num_innermost_ring_layers
+                 << std::endl;
+
+      // error message
+      if (!_quad_center_elements)
+      {
+        paramError("ring_block_ids",
+                   "This parameter must have the appropriate size if it is provided: "
+                   "If the first element of 'ring_intervals' is more than one, "
```

Comment:
it's not just that, it s also the size of the boundary layers.
if either:
- there is more than one interval in the innermost ring
- there is a boundary layer (inner or outer) in the innermost ring
and
- we are not using quad center elements
Then there must be an extra block id (for the non-quad element)

Reply:
The boundary layer and the ring intervals together make the num_innermost_ring_layers. I can rephrase it like that